### PR TITLE
Pass dockerhub token for authorized pulls

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/13981

### CODE REVIEW

**Review by whites11 — APPROVED** [review]
LGTMIIWFY

**Comment by fiunchinho** [comment]
Does this mean that we will pay DockerHub to be able to pull images? or authenticating using a free account give us bigger limits than anonymous pulls?

**Comment by corest** [comment]
@fiunchinho we pay dockerhub

File: service/controller/cloudconfig/cloud_config.go [hunk 1]
```diff
@@ -78,6 +80,7 @@ func New(config Config) (*CloudConfig, error) {
 
 		azure:                  config.Azure,
 		azureClientCredentials: config.AzureClientCredentials,
+		dockerhubToken:         config.DockerhubToken,
```
Comment (fiunchinho) [inline service/controller/cloudconfig/cloud_config.go:83]: should we check if the token is empty and fail in that case?

**Review by fiunchinho — APPROVED** [review]
for the CAPI/CAPZ versions we use the `spark` resource to generate the worker cloud config. You'd need to update it too. Also remember to update the changelog.

**Review by fiunchinho — CHANGES_REQUESTED** [review]
for the CAPI/CAPZ versions we use the spark resource to generate the worker cloud config. You'd need to update it too. Also remember to update the changelog.

**Review by fiunchinho — APPROVED** [review]
Ok, Vol just made me realize this is not targeting the master branch so no need to change the `spark` handler.

File: service/controller/cloudconfig/cloud_config.go (hunk 1 above)
Comment (corest) [reply service/controller/cloudconfig/cloud_config.go:83]: yes, added

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**